p2p/gossip.go

```diff
@@ -50,8 +53,7 @@ func NewGossiper(host host.Host, ps *pubsub.PubSub, topicStr string, msgHandler
 	if err != nil {
 		return nil, err
 	}
-
-	subscription, err := topic.Subscribe()
+	subscription, err := topic.Subscribe(pubsub.WithBufferSize(max(pubsub.GossipSubHistoryGossip, pubsubBufferSize)))
```

Comment:
not sure why we have `GossipSubCacheSize: 50` default, if take a max with `pubsubBufferSize = 3000`.
Why not setting the default to be 3000 and just use the `GossipSubCacheSize` conf param?

Reply:
if u wish to enforce it, u can check it in 
`(c P2PConfig) Validate()`, but I think it's unneeded

Reply:
i dont think is a good idea to set it to the same value, since GossipSubCacheSize has another (network) implications, such a periodic overhead that nodes need to send to advertise cached packets, while pubsubBufferSize its just affecting the local buffer size for non-consumed packets. Ideally, we want to keep GossipSubCacheSize smaller than pubsubBufferSize.